org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -113,6 +114,8 @@
 
     private ZigBeeNetworkMeshMonitor meshMonitor = null;
 
+    private boolean bridgeRemoved = false;
```

Comment:
This variable is set in method `dispose` and in method `handleRemoval`. As far as I understood you in personal communication, these two methods might be called concurrently. If this is really the case, this variable should be `volatile`.

Reply:
`handleRemoval` is called first and the first instruction there sets this variable. When `super.handleRemoval` is invoked then `dispose` will be executed in parallel. Not sure if this could be a potential issue but yes, `volatile`
 makes it more save.